scripts/inference/hf_generate.py

```diff
@@ -58,12 +60,25 @@ def parse_args() -> Namespace:
         default=[
             'My name is',
             'This is an explanation of deep learning to a five year old. Deep learning is',
-        ])
-    parser.add_argument('--max_new_tokens', type=int, default=100)
+        ],
+        help='Generation prompts. Use syntax "file::/path/to/prompt.txt" to load a ' +\
+             'prompt contained in a txt file.'
+        )
     parser.add_argument('--max_seq_len', type=int, default=None)
-    parser.add_argument('--temperature', type=float, default=1.0)
-    parser.add_argument('--top_k', type=int, default=50)
-    parser.add_argument('--top_p', type=float, default=1.0)
+    parser.add_argument('--max_new_tokens', type=int, default=100)
+    parser.add_argument('--max_batch_size', type=int, default=None)
+    parser.add_argument('--temperature', type=float, nargs='+', default=[1.0])
+    parser.add_argument('--top_k', type=int, nargs='+', default=[50])
```

Comment:
Do you know if `--top_k 50` is a good default / the same one that HF generate uses? I put it there because I read it somewhere in a guide,  but then someone said that it was a weird settting. Maybe we should consider reverting to `None`?

Similarly for `top_p` and `temperature`, but I think those are set to behave as no-op right now

Reply:
If we wanted to use the defaults set by HF (see here https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationConfig), we'd set `--top_k 50` (which is surprising, kinda) and `--top_p 1.0`.

I'm in support of setting all of these generation knobs to the HF defaults, so that if the user doesn't set them they get the default behavior, but also we'd be exposing those defaults through the arg parser. I think we'd only need to change `top_p`.

Reply:
Scratch that, they already match the HF defaults.